internal/node/hdb/state/state_machine.go

```diff
@@ -67,36 +54,20 @@ func (sm *StateMachine) StartListening(ctx context.Context) {
 			return
 		// TODO this bit of code should be well tested
 		case stateUpdate := <-sm.updateChan:
+			// execute state update
+			stateBytes, err := stateUpdate.NewState().Bytes()
+			if err != nil {
+				log.Error().Err(err).Msgf("error getting new state bytes from state update chan")
+			}
+			jsonState, err := hdb.NewJSONState(sm.schema, stateBytes)
+			if err != nil {
+				log.Error().Err(err).Msgf("error getting new state from state update chan")
```

Comment:
this really shouldn't be a log + move on error, it should return something to the caller
should go away with state machine removal